src/components/colorscale/attributes.js

```diff
@@ -160,6 +162,21 @@ module.exports = function colorScaleAttrs(context, opts) {
         ].join('')
     };
 
+    attrs[mid] = {
+        valType: 'number',
+        role: 'info',
+        dflt: null,
+        editType: editTypeOverride || 'plot',
```

Comment:
@etpinard I wasn't sure what to put for `editType` and `impliedEdits` btw so I copied over the settings from `max` above... maybe it should match `auto` instead? Or be something else? Likely your tests will expose any problems but I just wanted to shine a light here.

Reply:
Restyle got fixed in https://github.com/plotly/plotly.js/pull/3549/commits/23c1cf0832a585a47b4231dbd9e9adfd785d382a

Is that the behaviour you had in mind?

Reply:
I must admit I don't fully grok the various nuances of the types of edits, but the test reads fine at first glance!